### PR TITLE
Data Union contracts that allow specifying a custom token, among other things

### PR DESCRIPTION
**Pull request by jtakalai**
From the original PR #73 

- Update solidity compiler to 0.8.6 for all contracts
- Remove SafeMath library which is not needed anymore for solidity compiler >= 0.8
- Remove Mainnet/SideChain migration manager contracts
- Add token and mediator parameters to 'initialize' function of Mainnet/SideChain DataUnion contracts
- Add duFee and duBeneficiary parameters and their relevant functions to the Mainnet DataUnion contract

### CODE REVIEW

File: packages/data-union-solidity/contracts/BinanceAdapter.sol [hunk 1]
```diff
@@ -53,11 +54,11 @@ contract BinanceAdapter {
 
     function setBinanceRecipientFromSig(address from, address recipient, bytes memory sig) public {
         UserData storage userdata = binanceRecipient[from];
-        uint nextNonce = userdata.nonce.add(1);
+        uint nextNonce = ++(userdata.nonce);
         require(getSigner(recipient, nextNonce, sig) == from, "bad_signature");
         userdata.nonce = nextNonce;
```
Comment (Isentrope1) [inline packages/data-union-solidity/contracts/BinanceAdapter.sol:59]: is this line needed? you incremented userdata.nonce already

Comment (jtakalai) [reply packages/data-union-solidity/contracts/BinanceAdapter.sol]: 👍

**Review by samt1803 — APPROVED** [review]
looked it over, looks good! didn't run it or run tests though. Only CI run needs to be fixed

File: packages/data-union-solidity/contracts/BinanceAdapter.sol [hunk 2]
```diff
@@ -87,25 +88,26 @@ contract BinanceAdapter {
         }
         toCoin.transferAndCall(bscBridge, sendToBinanceAmount, BytesLib.toBytes(binanceAddress));
         emit WithdrawToBinance(address(toCoin), binanceAddress, amountDatacoin, sendToBinanceAmount);
-        datacoinPassed = datacoinPassed.add(amountDatacoin);
+        datacoinPassed = datacoinPassed + amountDatacoin;
```
[the remark this answers is not in the document]
Comment (jtakalai) [reply packages/data-union-solidity/contracts/BinanceAdapter.sol:90]: because safemath isn't required in Solidity 0.8 anymore

**Comment by jtakalai** [comment]
This was all merged with #77

**Pull request closed without merging**